backend/src/api/workspaces.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express')
+const { Workspace } = require('../models')
+const { directorsOnly, errorWrap } = require('../middleware')
+const router = express.Router()
+
+// Get all workspaces
+// TODO: when authentication server is integrated, only show the workspace that
+// belong to the caller of this endpoint.
+router.get(
+  '/',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    const workspaces = await Workspace.find()
+    res.json({
+      code: 200,
+      result: workspaces,
+      success: true
+    })
+  })
+)
+
+// Find workspace by name
+// TODO: when authentication server is integrated, only show the workspace if it
+// belongs to the caller of this endpoint.
+router.get(
+  '/:workspaceName',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    const workspaceName = req.params.workspaceName
+
+    if (!workspaceName) {
+      return res.json({
+        code: 400,
+        message: 'malformed request',
+        success: false
+      })
+    }
+
+    const workspace = await Workspace.find({ name: workspaceName })
+
+    res.json({
+      code: 200,
+      result: workspace,
+      success: true
+    })
+  })
+)
+
+// Create a new workspace
+router.post(
+  '/',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    const owner = req.body.owner
+    const workspaceName = req.body.name
+    if (!owner || !workspaceName) {
+      return res.json({
+        code: 400,
+        message: owner ? 'Invalid owner name' : 'Invalid workspace name',
```

Comment:
i think this should be the other way around (if owner exists the msg is 'invalid workspace')

Reply:
lolol yeah good catch